### DISCUSSION TITLE
Calling rcl_parse_arguments

### DISCUSSION THREAD

**Discussion by jimmy-mcelwain** [Yaskawa-Global/motoros2#327] [Q&A]
Is there a particular reason that we are calling `rcl_parse_arguments` directly in `CommunicationExecutor.c`? Why don't we just pass in the `faux_arg`c and `faux_argv` to `rclc_support_init_with_options`? `rcl_parse_arguments` is (eventually) called there, and the remap rules still work properly. Is it just so that it doesn't enter the error loop forever if the user has invalid remap rules? Is there some other reason?  If so, I understand, but I wasn't sure if this is an oversight or a intentional decision.

**Comment by gavanderhoorn** [comment]
The main reason -- but I'd have to look up my notes from way back when that was implemented -- most likely is we wanted to use the smallest part of RCL and RCLC which would get us remap functionality.

So keep the existing control flow, then add the minimum needed to implement remapping support.

We could refactor and call 

https://github.com/Yaskawa-Global/motoros2/blob/0f0a510b4989afaf8925ef174784cbd26df462da/src/CommunicationExecutor.c#L140-L141

later, after having created the faux `argv` and `argc`. Functionally, it wouldn't really change anything IIUC.

Priority of other issues/enhancement requests was just always higher.


PS: please always link to the code you're referring to. Makes discussing it easier :)
